inst/app/R/fct_dea-boostrap.R

```diff
@@ -0,0 +1,94 @@
+bw_rule <- function(delta, rule = 'cv', adjust = TRUE, type = NULL) {
+  bw_rule <- 3
+  if (is.character(rule)) {
+    bw_rule <- switch(rule,
+                      'silverman' = 1,
+                      'scott' = 2,
+                      'cv' = 3,
+                      'sw98' = 4,
+                      3
+    )
+    if (is.null(bw_rule)) {
+      cli::cli_warn('Unknown bandwidth rule. Using default value instead')
+    }
+  } else if (is.numeric(rule)) {
+    if (rule %in% 1:3) {
+      bw_rule <- rule
+    } else {
+      cli::cli_warn('Unknown bandwidth rule. Using default value instead')
+    }
+  } else {
+    cli::cli_warn('Unknown bandwidth rule. Using default value instead')
+  }
+  # Values must be in range 1, Inf. Take inverse if values are in range 0, 1
+  if (min(delta) < 1) {
+    delta <- 1/delta
+  }
+  # Values == 1 are artifacts of DEA method and should be removed for h
+  delta_m <- delta[delta > 1]
+  delta_2m <- c(delta_m, 2 - delta_m)
```

Comment:
Almost the same code as in `bootstrap_sample()` . Could consider a helper function , e.g.`get_delta2m()`, with a truncate/remove TRUE/FALSE parameter?

Reply:
I've looked at this. While this is not completely DRY, the number of lines we are going to save compared to the added complexity and reduced readability of adding a function, it not worth it in my opinion. If we had the tuple available as in Python, we could return both `delta` and `delta_2m` and do the check for `min(delta) < 1` as well. That would make more sense.

Reply:
I agree that the choice here is not obvious. Let's keep this as is for now, and then revisit it in pear.